platform/mellanox/cmis_host_mgmt/cmis_host_mgmt.py

```diff
@@ -0,0 +1,151 @@
+            lines = param_file.read()
+
+        if lines:
+            if action == "disable":
+                lines = re.sub(CMISHostMgmtActivator.PARAMS[param]["enabled_param"],
+                               CMISHostMgmtActivator.PARAMS[param]["disabled_param"],
+                               lines)
+            elif action == "enable":
+                if param == "sai_profile" and not re.search(CMISHostMgmtActivator.PARAMS[param]["disabled_param"], lines):
+                    with open(file_path, 'a') as param_file:
+                        param_file.write(CMISHostMgmtActivator.PARAMS[param]["enabled_param"])
+                    return
+
+                lines = re.sub(CMISHostMgmtActivator.PARAMS[param]["disabled_param"],
+                               CMISHostMgmtActivator.PARAMS[param]["enabled_param"],
+                               lines)
+
+        with open(file_path, 'w') as param_file:
+            param_file.write(lines)
+
+
+    @staticmethod
+    def parse_show_platform_summary():
+        summary = subprocess.check_output(['show', 'platform', 'summary'])
+        summary = summary.decode('utf-8')
+        summary = [x for x in summary.split('\n') if x]
+
+        for field in summary:
+            key, value = field.split(": ")
+            
+            if key == 'Platform':
+                platform = value
+
+            elif key == 'HwSKU':
+                sku = value
+
+        return platform, sku
+
+
+    @staticmethod
+    def remove_im_file(file_path):
+        if os.path.isfile(file_path):
+            os.remove(file_path)
+
+
+    @staticmethod
+    def copy_im_file(src_path, dest_path):
+        if os.path.isfile(src_path):
+            shutil.copy(src_path, dest_path)
+
+
+    @staticmethod
+    def disable_im():
+        platform, sku = CMISHostMgmtActivator.parse_show_platform_summary()
+        sku_path = '/usr/share/sonic/device/{0}/{1}'.format(platform, sku)
+        platform_path = '/usr/share/sonic/device/{0}'.format(platform)
+        CMISHostMgmtActivator.change_param("sai_profile", sku_path, 'disable')
+        CMISHostMgmtActivator.change_param("pmon_daemon_control", platform_path, 'disable')
+        
+        CMISHostMgmtActivator.remove_im_file('{0}/{1}'.format(sku_path, 'media_settings.json'))
+        CMISHostMgmtActivator.remove_im_file('{0}/{1}'.format(sku_path,'optics_si_settings.json'))
+        CMISHostMgmtActivator.remove_im_file('{0}/{1}'.format(platform_path, 'media_settings.json'))
+        CMISHostMgmtActivator.remove_im_file('{0}/{1}'.format(platform_path, 'optics_si_settings.json'))
+        CMISHostMgmtActivator.remove_im_file('{0}/{1}'.format(sku_path, 'pmon_daemon_control.json'))
+
+
+    @staticmethod
+    def enable_im(args):
+        platform, sku = CMISHostMgmtActivator.parse_show_platform_summary()
+        sku_path = '/usr/share/sonic/device/{0}/{1}'.format(platform, sku)
+        platform_path = '/usr/share/sonic/device/{0}'.format(platform)
+
```

Comment:
@skr31 @dprital
do we want to have some kind of platform limitation? 
Or not, so we won't have to update the script each time we add support to a new platform? 

My suggestion is at least not approve the enablement of the feature is the platform is lower that spc3.
If the user tries to do it, need to print some error and return.

Reply:
Added check for spc3 and higher for now